### PR TITLE
SDK-578: Create transition resolver

### PR DESCRIPTION
**Pull request by vol4onok**
https://spryker.atlassian.net/browse/SDK-578

### CODE REVIEW

File: src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php [hunk 1]
```diff
@@ -38,6 +55,11 @@ class WorkflowTransitionListener
      */
     public const META_KEY_TASK = 'task';
 
+    /**
+     * @var string
+     */
+    public const META_KEY_TRANSITION_RESOLVER = 'transition_resolver';
```
Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php:61]: Please don't mix camelCase and snake_case for workflow metadata

File: src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php [hunk 2]
```diff
@@ -28,6 +30,21 @@
 
 class WorkflowTransitionListener
 {
+    /**
+     * @var string
+     */
+    public const DATA_NEXT_TRANSACTION = 'nextTransition';
```
Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php:36]:
```suggestion
    public const DATA_NEXT_TRANSITION = 'nextTransition';
```

File: src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php [hunk 3]
```diff
@@ -152,7 +185,36 @@ protected function tryRunTask(TransitionEvent $event, WorkflowTransitionInterfac
...
+        );
+    }
+
+    /**
+     * @param \Symfony\Component\Workflow\Event\TransitionEvent $event
+     * @param \SprykerSdk\SdkContracts\Entity\ContextInterface $context
+     *
+     * @return string|null
+     */
+    protected function resolverNextTransaction(TransitionEvent $event, ContextInterface $context): ?string
```
Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php:207]:
```suggestion
    protected function resolverNextTransition(TransitionEvent $event, ContextInterface $context): ?string
```

File: src/Infrastructure/Service/WorkflowRunner.php [hunk 4]
```diff
@@ -136,4 +142,26 @@ protected function getNextTransition(ProjectWorkflow $projectWorkflow): ?string
...
+    /**
+     * @param \SprykerSdk\Sdk\Core\Appplication\Service\ProjectWorkflow $projectWorkflow
+     * @param array<string> $nextEnabledTransitions
+     *
+     * @return string|null
+     */
+    protected function getPreDefinedTransition(ProjectWorkflow $projectWorkflow, array $nextEnabledTransitions): ?string
+    {
+        $runningTransition = $projectWorkflow->getPreviousTransition();
+        if ($runningTransition) {
```
Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Service/WorkflowRunner.php:155]: Please remove nested conditions and make early return

File: src/Core/Appplication/Service/ProjectWorkflow.php [hunk 5]
```diff
@@ -255,6 +255,34 @@ public function getRunningTransition(?WorkflowInterface $workflow = null): ?Work
...
     }
 
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\WorkflowInterface|null $workflow
+     *
+     * @throws \SprykerSdk\Sdk\Core\Appplication\Exception\ProjectWorkflowException
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\WorkflowTransitionInterface|null
+     */
+    public function getPreviousTransition(?WorkflowInterface $workflow = null): ?WorkflowTransitionInterface
```
Comment (Kyrylo-Meretskyi) [inline src/Core/Appplication/Service/ProjectWorkflow.php:265]:
```suggestion
    public function findPreviousTransition(?WorkflowInterface $workflow = null): ?WorkflowTransitionInterface
```

**Review by Kyrylo-Meretskyi — APPROVED** [review]

**Review by miroshnikov-spryker — APPROVED** [review]

**Pull request merged**